Got AssertionError when receive file

Hey, I tried to use the latest version, 0.4.4, to transfer files between nova instance and host, mainly instance to host. I've redirected the instance's serial port to host's unix domain socket.
On the instance  
```
import serial

ser = serial.Serial(port = '/dev/ttyS2',baudrate=9600,timeout=3)
ser.flushInput()
def getc(size,timeout=1):
    data = ser.read(size)
    return data

def putc(data,timeout=1):
    return ser.write(data)

from xmodem import XMODEM
modem = XMODEM(getc,putc)
test = file('test.txt','rb')
print modem.send(test)
ser.close()
```  
On host  
```
import socket
import sys

if len(sys.argv)!=2:
    print 'usage:python '
unix_socket = sys.argv[1]
s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
s.connect(unix_socket)
def getc(size,timeout=1):
    data = s.recv(size)
    return data

def putc(data,timeout=1):
    return s.send(data)

from xmodem import XMODEM
modem = XMODEM(getc,putc)
test = file('test.txt','wb')
print modem.recv(test)
s.close()
```  

Then I ran these two scripts, the guest send a file and the host try to receive it. But the host constantly got errors and then received the file finally.
```
root@node-51:~# python receivefile.py /tmp/virtualsocket/0a125fca-efac-4b4d-b681-7c78f3087713 
No handlers could be found for logger "xmodem.XMODEM"
Traceback (most recent call last):
  File "receivefile.py", line 19, in <module>
    print modem.recv(test)
  File "build/bdist.linux-x86_64/egg/xmodem/__init__.py", line 531, in recv
AssertionError: h
root@node-51:~# python receivefile.py /tmp/virtualsocket/0a125fca-efac-4b4d-b681-7c78f3087713 
No handlers could be found for logger "xmodem.XMODEM"
Traceback (most recent call last):
  File "receivefile.py", line 19, in <module>
    print modem.recv(test)
  File "build/bdist.linux-x86_64/egg/xmodem/__init__.py", line 531, in recv
AssertionError:  
root@node-51:~# python receivefile.py /tmp/virtualsocket/0a125fca-efac-4b4d-b681-7c78f3087713 
No handlers could be found for logger "xmodem.XMODEM"
Traceback (most recent call last):
  File "receivefile.py", line 19, in <module>
    print modem.recv(test)
  File "build/bdist.linux-x86_64/egg/xmodem/__init__.py", line 531, in recv
AssertionError:  
root@node-51:~# python receivefile.py /tmp/virtualsocket/0a125fca-efac-4b4d-b681-7c78f3087713 
No handlers could be found for logger "xmodem.XMODEM"
Traceback (most recent call last):
  File "receivefile.py", line 19, in <module>
    print modem.recv(test)
  File "build/bdist.linux-x86_64/egg/xmodem/__init__.py", line 531, in recv
AssertionError:  
root@node-51:~# python receivefile.py /tmp/virtualsocket/0a125fca-efac-4b4d-b681-7c78f3087713 
No handlers could be found for logger "xmodem.XMODEM"
Traceback (most recent call last):
  File "receivefile.py", line 19, in <module>
    print modem.recv(test)
  File "build/bdist.linux-x86_64/egg/xmodem/__init__.py", line 531, in recv
AssertionError:  
root@node-51:~# python receivefile.py /tmp/virtualsocket/0a125fca-efac-4b4d-b681-7c78f3087713 
No handlers could be found for logger "xmodem.XMODEM"
Traceback (most recent call last):
  File "receivefile.py", line 19, in <module>
    print modem.recv(test)
  File "build/bdist.linux-x86_64/egg/xmodem/__init__.py", line 531, in recv
AssertionError:  
root@node-51:~# python receivefile.py /tmp/virtualsocket/0a125fca-efac-4b4d-b681-7c78f3087713 
No handlers could be found for logger "xmodem.XMODEM"
Traceback (most recent call last):
  File "receivefile.py", line 19, in <module>
    print modem.recv(test)
  File "build/bdist.linux-x86_64/egg/xmodem/__init__.py", line 531, in recv
AssertionError:  
root@node-51:~# python receivefile.py /tmp/virtualsocket/0a125fca-efac-4b4d-b681-7c78f3087713 
128
root@node-51:~# 
```  
I checked the xmodem/__init__.py line 531
```
assert False, data
```  
Did I make a mistake or is this a bug?

--- a/xmodem/__init__.py
+++ b/xmodem/__init__.py
@@ -560,7 +560,6 @@ class XMODEM(object):
                 data = self.getc(1, timeout=1)
                 if data is None:
                     break
-                assert False, data
             self.putc(NAK)
             # get next start-of-header byte
             char = self.getc(1, timeout)

